mlflow/pyfunc/__init__.py

```diff
@@ -439,26 +560,41 @@ def spark_udf(spark, model_uri, result_type="double"):
 
     def predict(*args):
         model = SparkModelCache.get_or_load(archive_path)
-        schema = {str(i): arg for i, arg in enumerate(args)}
+        input_schema = model.metadata.get_input_schema()
         pdf = None
+
         for x in args:
             if type(x) == pandas.DataFrame:
                 if len(args) != 1:
                     raise Exception("If passing a StructType column, there should be only one "
                                     "input column, but got %d" % len(args))
                 pdf = x
         if pdf is None:
-            # Explicitly pass order of columns to avoid lexicographic ordering (i.e., 10 < 2)
-            columns = [str(i) for i, _ in enumerate(args)]
-            pdf = pandas.DataFrame(schema, columns=columns)
+            args = list(args)
+            if input_schema is None:
+                names = [str(i) for i in range(len(args))]
+            else:
+                names = input_schema.column_names()
```

Comment:
What is the value of column_names if the schema provided no column names?

Reply:
range from 0..num_cols